### PR TITLE
Update Readme.md

### PR DESCRIPTION
**Pull request by Mouhajer-CO**
### JIRA link

Resolves [NTRNL-370](https://technologyprogramme.atlassian.net/browse/NTRNL-370)

### Description

Update Readme.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,115 @@
-# node-login
-Node login module
+# Login
+
+<!-- ![npm](https://img.shields.io/npm/v/%40co-digital%2Flogin) -->
+![Static Badge](https://img.shields.io/badge/test_coverage-%E2%89%A595%25-green)
+
+## Overview and Scope
+
+This `@co-digital/login` aims to provide a simple solution for implementing user authentication within your Node.js applications. With a focus on security and ease of integration, our solution incorporates a middleware that connects to our internal Identity provider COLA and relatively soon OneLogin.
```
Comment (Harrisman05) [inline README.md:8]: Remove 'your' - instead: `within CO Node.js applications` ?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,115 @@
...
+
+Key             |  Description               | Example Value
+----------------|--------------------------- |-------------------------
+USER_POOL_ID | ID of the user pool in Amazon Cognito | `secret`
+USER_POOL_CLIENT_ID | Client ID of an app registered with the user pool in Amazon Cognito | `secret`
+AUTH_SIGN_IN_URL | Authentication sign in URL | `https://cola.service.cabinetoffice.gov.uk/v2/<YOUR_SERVICE>/login`
+COOKIE_ID_NAME | The name of the cookie | `github-requests` for [github-requests-app](https://github.com/cabinetoffice/github-requests-app)
+COOKIE_PARSER_SECRET | Secret used in validating/calculating the cookie signature | `secret`
+COOKIE_SESSION_SECRET | Secret key for signing the session cookie | `secret`
+LOG_LEVEL       | Logging levels        | `info` (Log only if info.level is less than or equal to this level)
```
Comment (Harrisman05) [inline README.md:28]: wrap info.level in backticks

File: README.md [hunk 3]
```diff
@@ -1,2 +1,115 @@
...
+HUMAN | Formatting messages form (default JSON) | `true` (Enable human formatting for log messages)
+
+## Files Structure
+
+Directory Path | Description
+--- | ---
+`./config/` | This folder contains configuration settings for the login module. It is essential for fetching sensitive information such as `COOKIE_PARSER_SECRET` or `COOKIE_SESSION_SECRET`.
+`./util/` | This folder contains utility functions used throughout the login module, including Session and Cookie utils.
```
Comment (Harrisman05) [inline README.md:36]: missing s in utils

File: README.md [hunk 4]
```diff
@@ -1,2 +1,115 @@
...
+`./index.ts` | This file will expose the authentication middleware and Session and Cookie utils methods.
```
Comment (Harrisman05) [inline README.md:37]: remove 'methods' for consistency with sentence above

File: README.md [hunk 5]
```diff
@@ -1,2 +1,115 @@
...
+
+```ts
+// On app.ts
+...
+app.use(cookieParser(COOKIE_PARSER_SECRET));
+app.use(cookieSession({ secret: COOKIE_SESSION_SECRET }));
+...
+```
+
+- **Route Setup**: Define routes for any authentication-related endpoints required by your application and utilize the middleware to protect routes that require authentication.
```
Comment (Harrisman05) [inline README.md:61]: -ise suffix

File: README.md [hunk 6]
```diff
@@ -1,2 +1,115 @@
...
+// On any endpoint when auth is required (eg. home route)
+...
+import { authentication } from "@co-digital/login";
+
+const homePageRouter = Router();
+homePageRouter.get(config.HOME_URL, authentication, get);
+...
+```
+
+- **Frontend Integration**: `COLA` has already developed user interface components for login, logout, and other authentication-related actions. However, for integration with other Identity Providers (IdPs) such as `OneLogin`, you'll need to implement these components within your application.
```
Comment (Harrisman05) [inline README.md:73]: One Login is 2 separate words, see example in their docs https://github.com/govuk-one-login/authentication-tech-docs

File: README.md [hunk 7]
```diff
@@ -1,2 +1,115 @@
...
+
+## Note
+
+- `COLA` provides a secure login mechanism for any Cabinet Office service that hosts under the some domain - `cabinetoffice.gov.uk` and under the hood utilises and abstracts AWS Cognito for user management.
```
Comment (Harrisman05) [inline README.md:77]: Unclear wording/structure, something like:

"... service that is hosted under the `cabinetoffice.gov.uk` domain. Under the hood, it utilises and abstracts AWS Cognito for user management."

File: README.md [hunk 8]
```diff
@@ -1,2 +1,115 @@
...
+
+- `cookieParser` method from `cookie-parser` module parse cookie header and populate `req.cookies` with an object keyed by the cookie names. The middleware will parse the cookie header on the request and expose the cookie data as the property `req.signedCookies` when a secret was provided.
```
Comment (Harrisman05) [inline README.md:79]: parses

File: README.md [hunk 9]
```diff
@@ -1,2 +1,115 @@
...
+
+- `cookieParser.signedCookie(str, secret)` parse a cookie value as a signed cookie. This will return the parsed unsigned value if it was a signed cookie and the signature was valid. If the value was not signed, the original value is returned. If the value was signed but the signature could not be validated, false is returned.
```
Comment (Harrisman05) [inline README.md:81]: First sentence - "parses"

Last sentence - should `false` be in backticks?

File: README.md [hunk 10]
```diff
@@ -1,2 +1,115 @@
...
+
+- After the server sends a cookie to a client, the client might modify the cookie before sending it back to the server in a future request. So you sign a cookie if you want assurance that the data being returned in the cookie has not been modified by the client.
+
+- `cookieSession` creates a new cookie session middleware. This middleware will attach the property session to req, which provides an object representing the loaded session. This session is either a new session if no valid session was provided in the request, or a loaded session from the request. The `req.session` represents the session for the given request.
```
Comment (Harrisman05) [inline README.md:85]: `req` in backticks?

File: README.md [hunk 11]
```diff
@@ -1,2 +1,115 @@
...
+```ts
+app.use(
+  cookieSession({
+    // A string which will be used as single key if keys is not provided.
+    secret: process.env.COOKIE_SESSION_SECRET,
+  })
+);
+```
+
+### Why cookie-session module
```
Comment (Harrisman05) [inline README.md:96]: Purpose of cookie-session module

File: README.md (hunk 6 above)
Comment (Mouhajer-CO) [reply README.md:73]: Changed to `GOV.UK One Login`

**Review by Harrisman05 — APPROVED** [review]

**Pull request merged**